### DISCUSSION TITLE
What does `linen.transforms.scan` do?

### DISCUSSION THREAD

**Discussion by changjonathanc** [google/flax#1067] [Q&A] [answered]
I saw it here:
https://github.com/google/flax/blob/93078a428b5e4abc5c516c6b24b07fa885f22c5a/examples/seq2seq/train.py#L157-L164
And it's defined here with a TODO:
https://github.com/google/flax/blob/93078a428b5e4abc5c516c6b24b07fa885f22c5a/flax/linen/transforms.py#L198-L202

**Comment by avital** [comment]
I believe that @jheek was planning on adding clear docstrings to these.

**Comment by jheek** [comment, marked as answer]
These are the "lifted" jax transformations for linen. The short story is that jax transforms don't work natively with stateful things (eg. variables, rng sequences) like linen Modules. These lifted transforms wrap the existing jax transforms and add functionality on top it to support modules fully. You can check `flax.core.lift` for their implementation and a short docstring. Over the last months we have stabilised the implementation and you can soon expect proper documentation and examples for these features.
